x/mongo/driver/connstring/connstring.go

```diff
@@ -299,6 +301,24 @@ func (p *parser) parse(original string) error {
 		if err != nil {
 			return err
 		}
+
+		// If p.SRVMaxHosts is non-zero, randomly select up to SRVMaxHosts hosts from
+		// parsedHosts using the modern Fisher-Yates algorithm.
```

Comment:
The [Fisher-Yates modern algorithm](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm) should be an `O(n)` way of shuffling an array. I implemented the pseudocode described in the "equivalent version" that shuffles from lowest to highest index.

Reply:
Can we use [`rand#Shuffle`](https://pkg.go.dev/math/rand#Shuffle) here instead?